### ISSUE TITLE
Bug: Default submit script is overridden by `RemoteData`

### ISSUE THREAD

**Issue by hujay2019** [sphuber/aiida-shell#58]
The error occurs when the following conditions are met:

1. `submit=True`. Run a script in a non-blocking way. Then a correct script `_aiidasubmit.sh` will be written in the working directory.
2. Add a `RemoteData` type node as input. If the `RemoteData` contains the file `_aiidasubmit.sh` generated by the previous process, this 'wrong' script will override the correct script.

This error can be avoided by defining a different submit script filename (see examples below).
I suggest adding a WARNING to the documentation or changing the default behavior of the `aiida-shell' package to avoid this error.

Error example
```
results, future = launch_shell_job(
    "date",
    submit=True,
    nodes={"remote_folder": remote_folder},
    metadata={
        "options": {
            "computer": code.computer,
        }
    },
)
```

Correct example
```
results, future = launch_shell_job(
    "date",
    submit=True,
    nodes={"remote_folder": remote_folder},
    metadata={
        "options": {
            "computer": code.computer,
            "submit_script_filename": "_aiidasubmit_shell.sh",
        }
    },
)
```

**Comment by sphuber** [comment]
Thanks for the report and the work-around. The documentation actually _does_ warn about this currently. See the bottom of this section: https://aiida-shell.readthedocs.io/en/latest/howto.html#running-a-shell-command-with-remote-data
If you have any suggestions on how to improve the wording, that would be much appreciated.

I am wondering if instead of recommending this workaround, `aiida-shell` can do something to prevent this file from being overwritten. Unfortunately, it seems like that might be more tricky than one would think. I am using the `CalcInfo.remote_copy_list` to have the engine copy over the files to the work directory and those are copied _after_ the contents of the sandbox folder are copied over:

https://github.com/aiidateam/aiida-core/blob/main/aiida/engine/daemon/execmanager.py#L245-L249

The `remote_copy_list` syntax also does not allow any "exclude" rules, but only paths to _include_, so we cannot simply instruct the engine to skip `_aiidasubmit.sh`. Of course we could try to read the contents of the remote and manually add explicit include rules that skip the submit script. However, that requires opening a transport connection to the remote from within the `Calcjob.prepare_for_submission` method, which is ill-advised, since it can fail and it circumvents the engine's transport connection pooling.

There is actually already an open issue on `aiida-core` about this problem: https://github.com/aiidateam/aiida-core/issues/6012
I will discuss it with the team whether we can change the order of copying, which would solve this issue, but it is technically backwards-incompatible so I am not sure whether we can get this released soon.

**Comment by giovannipizzi** [comment]
I just had the same issue. Of course one thing is to avoid this to happen altogether, but I think it's a very common use case to just copy a subfolder from a parent remote to a subfolder of the new remote, but this is not possible AFAIK at the moment. People encounter this "bug" or behaviour because it's the first thing you try when you want to try to obtain this goal. Enabling this option already would have people try directly the subdirectory copy rather than a full copy. One could also try to warn, in a new version, if you are copying a RemoteData and don't specify a subfolder? or ask to ignore any files with the `"submit_script_filename"` name?

**Comment by giovannipizzi** [comment]
In terms of functionality for the remote data, I would suggest allowing to optionally specify a subdirectory or list of files/subdirs to copy from, default being `.` (with care between copying the content of a folder vs. the folder itself, i.e. with examples for the two cases), and a destination subdir (default `.` if there is a default mechanism to avoid clashes of stdout/stderr/status/submit script, otherwise the default could be to put in a subfolder? Still, it's still good to think of a mechanism to avoid copying at least those files). There should be a simple example to copy subfolder `mydir` to a subfolder with the same name in the new job working directory, as I think it is a potentially common use case.

**Comment by giovannipizzi** [comment]
> Ultimately, I am not sure what the best approach is here. I really want to avoid coming up with a similar syntax as remote_copy_list in aiida-core.

But in the end maybe it's better to have a simpler version that works in most cases (e.g. copy everything? but there are the issues described above), and still expos the full capability for those who need it? I still think anyway that copying only some files (directly on the remote) is a very important requirement for most usecases.

**Comment by sphuber** [comment]
> I still think anyway that copying only some files (directly on the remote) is a very important requirement for most usecases.

I agree.

> But in the end maybe it's better to have a simpler version that works in most cases (e.g. copy everything? but there are the issues described above)

That is what is currently implemented. [This PR](
https://github.com/sphuber/aiida-shell/pull/76) would fix those immediate issues and make it usable in most cases, but it requires a change in `aiida-core`. See this PR https://github.com/aiidateam/aiida-core/pull/6285 So it would be really useful if you can comment on that.

> and still expos the full capability for those who need it? 

You mean directly allow users to define `remote_copy_list`? I guess we could do that. But I think we should have a way to exclude files then as well when doing globbing, which is currently not supported. Maybe we should add this to `aiida-core` first?

**Comment by giovannipizzi** [comment]
Yes, probably putting this feature in aiida core first is the right thing to do, it's a useful use case

**Comment by sphuber** [comment]
@hujay2019 the fix for this is now released in v0.7.2 Thanks again for reporting
